vars/main.yml

```diff
@@ -11,3 +11,12 @@ _node_exporter_version_all: "{{ node_exporter_version.split('.') }}"
 _node_exporter_collectors_as_string_version: "{{ _node_exporter_version_all[0] | int == 0 and _node_exporter_version_all[1] | int < 15 }}"
 _node_exporter_command_version_check: "node_exporter {{ _node_exporter_collectors_as_string_version | ternary('-','--') }}version"
 _node_exporter_config_param_selector: --
+
+_node_exporter_arch_map:
+  i386: '386'
+  x86_64: 'amd64'
+  aarch64: 'arm64'
+  armv7l: 'armv7'
+  armv6l: 'armv6'
+
+node_exporter_arch: "{{ _node_exporter_arch_map[ansible_architecture] | default(ansible_architecture) }}"
```

Comment:
I'd remove the default fallback and add the remaining architectures in the mapping. Relying in the coincidence of the architecture name given by both ansible and node-exporter could be problematic